odiglet/pkg/kube/runtime_details/shared.go

```diff
@@ -65,32 +64,37 @@ func runtimeInspection(pods []corev1.Pod) ([]common.LanguageByContainer, error)
 				continue
 			}
 
-			detectionResults := inspectors.DetectLanguage(processes)
-			if len(detectionResults) == 0 {
-				log.Logger.V(0).Info("no supported language detected for container in pod", "processes", processes, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
-				continue
-			}
-
-			if len(detectionResults) > 1 {
-				log.Logger.V(0).Info("multiple languages detected for pod container processes, selecting first one", "processes", processes, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
-			}
-
-			resultsMap[container.Name] = common.LanguageByContainer{
-				ContainerName: container.Name,
-				Language:      detectionResults[0].Language,
+			for _, process := range processes {
+				lang, err := inspectors.DetectLanguage(process)
+				if err != nil {
+					log.Logger.V(0).Info("no supported language detected for container in pod", "process", process, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
+					continue
+				}
+
+				// Convert map to slice for k8s format
+				envs := make([]odigosv1.EnvVar, 0, len(process.Envs))
```

Comment:
I think requesting a capacity of `len(process.Envs)` is a bit wasteful, as the process can have dozens of env variables but we will at most take just one currently

Reply:
The process.Envs do not contain all the env vars defined for the process, only those we defined in `envOverwrite.EnvValues`
The actual filtering is done in `getEnvVars` in `process.go`.
* Do you think it worth renaming Envs to somethin like InstrumentationEnvs ?
* Is it better for the filtering to happen in the lower function (as it is currrently) or here?